internal/backend/azure/azure.go

```diff
@@ -184,22 +196,38 @@ func (be *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindRe
 
 	be.sem.GetToken()
 
-	debug.Log("InsertObject(%v, %v)", be.container.Name, objName)
+	debug.Log("InsertObject(%v, %v)", be.cfg.AccountName, objName)
 
 	var err error
-	if rd.Length() < 256*1024*1024 {
+	if rd.Length() < saveLargeSize {
 		// wrap the reader so that net/http client cannot close the reader
 		// CreateBlockBlobFromReader reads length from `Len()``
-		dataReader := azureAdapter{rd}
 
 		// if it's smaller than 256miB, then just create the file directly from the reader
-		ref := be.container.GetBlobReference(objName)
-		ref.Properties.ContentMD5 = base64.StdEncoding.EncodeToString(rd.Hash())
-		err = ref.CreateBlockBlobFromReader(dataReader, nil)
+		blockBlobClient := be.container.NewBlockBlobClient(objName)
+
+		// upload it as a new "block", use the base64 hash for the ID
+		id := base64.StdEncoding.EncodeToString(rd.Hash())
+
+		buf := make([]byte, saveLargeSize)
```

Comment:
The buffer is never filled with data. It should probably be:

```suggestion
		buf := make([]byte, rd.Length())
		_, err = io.ReadFull(rd, buf)
		if err != nil {
			return errors.Wrap(err, "ReadFull")
		}
```